### PR TITLE
WIP: Manage S3 buckets with Kingpin

### PR DESCRIPTION
**Pull request by diranged**

### CODE REVIEW

File: kingpin/actors/aws/s3.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    def _execute(self):
+        """Executes an actor and yields the results when its finished.
+
+        raises: gen.Return(True)
+        """
+        yield self._ensure_bucket()
+
+        # We don't go any further -- if we're deleting the bucket, we're done
+        # here. Just return.
+        if self.option('state') == 'absent':
```
Comment (siminm) [inline kingpin/actors/aws/s3.py]: Don't get the `if` here.

File: kingpin/actors/aws/s3.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+                raise exceptions.RecoverableActorFailure(
+                    'Cannot delete bucket: %s' % e.message)
+
+    @gen.coroutine
+    def _execute(self):
+        """Executes an actor and yields the results when its finished.
+
+        raises: gen.Return(True)
+        """
+        yield self._ensure_bucket()
```
Comment (siminm) [inline kingpin/actors/aws/s3.py]: If we're deleting a deleted bucket this actor does not produce any logs at all. Intentional?

**Comment by siminm** [comment]
few questions. otherwise looks good

File: kingpin/actors/aws/s3.py (hunk 1 above)
Comment (diranged) [inline kingpin/actors/aws/s3.py]: You'll see.. prep for the next commit.

File: kingpin/actors/aws/s3.py (hunk 2 above)
Comment (diranged) [inline kingpin/actors/aws/s3.py]: It actually does -- see line 196?

**Comment by diranged** [comment]
Going to merge this in for now -- I will have another commit shortly that adds bucket policy management.

**Pull request merged**